spec/Section 3 -- Type System.md

```diff
@@ -837,7 +837,6 @@ input ExampleInputObject {
 Original Value          | Variables       | Coerced Value
 ----------------------- | --------------- | -----------------------------------
 `{ a: "abc", b: 123 }`  | {null}          | `{ a: "abc", b: 123 }`
-`{ a: 123, b: "123" }`  | {null}          | `{ a: "123", b: 123 }`
```

Comment:
I think the fundamental problem here is that this section `Input coercion` mixes both aspects: "Input as ast literal" and "Input as variables (maps)".

For example in the first paragraph it states for the "Input as variables" aspect:
>  This unordered map should not contain any entries with names not defined by a field of this input object type, otherwise an error should be thrown.

But it doesn't say anything about extra fields for literals, because this is handled in the validation section. (I assume that is the reason)

In the table it is unclear if the "Original value" is a literal or a Map or both:
For example `{ b: $var }` can only be a literal, but the third entry `{ a: "abc" }` could be both.
And the second entry ` {a: 123, b: "123" }`  is wrong if it is a literal, because the types are wrong.
BUT: in the reference implementation it is actually correct if ` {a: 123, b: "123" }`  is interpreted as variable input. Could that be the reason it is there? (But I am not sure the reference implementation is correct in this case)